src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -387,6 +390,7 @@ public function iCompleteTheShippingStepWithFirstShippingMethod(): void
      * @When /^the (?:customer|visitor) proceed with ("[^"]+" payment)$/
      * @Given /^the (?:customer|visitor) has proceeded ("[^"]+" payment)$/
      * @When I try to change payment method to :paymentMethod payment
+     * @When I change payment method to :paymentMethod
```

Comment:
As I wrote in one of the comments in an earlier review iteration, it seems that in the API context some of the important behaviour is missing. [In UI](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ShopBundle/Resources/config/routing/order.yml#L42), after changing the payment method, payment takes place, while in API only the payment is changed.